koku/masu/processor/_tasks/data_validation.py

```diff
@@ -0,0 +1,222 @@
+    "OCPAzure": "reporting_ocpazurecostlineitem_project_daily_summary",
+    "OCPGCP": "reporting_ocpgcpcostlineitem_project_daily_summary",
+}
+
+
+class DataValidator:
+    """Class to check data is valid for providers"""
+
+    def __init__(
+        self,
+        schema,
+        start_date,
+        end_date,
+        provider_uuid,
+        ocp_on_cloud_type,
+        context,
+        date_step=settings.TRINO_DATE_STEP,
+    ):
+        self.schema = schema
+        self.provider_uuid = provider_uuid
+        self.ocp_on_cloud_type = ocp_on_cloud_type
+        self.start_date = start_date
+        self.end_date = end_date
+        self.context = context
+        self.date_step = date_step
+
+    def get_table_filters_for_provider(self, provider_type, trino=False):
+        """Get relevant table and query filters for given provider type"""
+        table_map = provider_type
+        if self.ocp_on_cloud_type:
+            table_map = f"OCP{provider_type}"
+        table = PG_TABLE_MAP.get(table_map)
+        query_filters = PG_FILTER_MAP.get(table_map)
+        if trino:
+            table = TRINO_TABLE_MAP.get(table_map)
+            query_filters = TRINO_FILTER_MAP.get(table_map)
+        return table, query_filters
+
+    def compare_data(self, pg_data, trino_data, tolerance=1):
+        """Validate if postgres and trino query data cost matches per day"""
+        incomplete_days = {}
+        valid_cost = True
+        if trino_data == {}:
+            return incomplete_days, False
+        for date in trino_data:
+            if date in pg_data:
+                if not abs(pg_data[date] - trino_data[date]) <= tolerance:
+                    incomplete_days[date] = {
+                        "pg_value": pg_data[date],
+                        "trino_value": trino_data[date],
+                        "delta": trino_data[date] - pg_data[date],
+                    }
+                    valid_cost = False
+            else:
+                incomplete_days[date] = "missing daily data"
+                valid_cost = False
+        return incomplete_days, valid_cost
+
+    def execute_relevant_query(self, provider_type, cluster_id=None, trino=False):
+        """Make relevant postgres or Trino queries"""
+        daily_result = {}
+        # year and month for running partitioned queries
+        dh = DateHelper()
+        year = dh.bill_year_from_date(self.start_date)
+        month = dh.bill_month_from_date(self.start_date)
+        report_db_accessor = ReportDBAccessorBase(self.schema)
+        # Set provider filter, when running ocp{aws/gcp/azure} checks we need to rely on the cluster id
+        provider_filter = self.provider_uuid if not self.ocp_on_cloud_type else cluster_id
+        # query trino/postgres
+        table, query_filters = self.get_table_filters_for_provider(provider_type, trino)
+        for start, end in date_range_pair(self.start_date, self.end_date, step=self.date_step):
```

Comment:
We currently use start and end dates based on the reports running. Their is a question here whether we should always run full month OR +x number of days since it _could_ be something outside the currently processing range that could be incorrect. Maybe this is ok for the first iteration though.

Reply:
Decided to add a range of -5 days to the validation checks, but not to cross the month boundary. This is also configurable via settings.